src/bitmap/backend/slice.rs

```diff
@@ -48,31 +64,26 @@ impl<'a, B: Bitmap> Bitmap for RefSlice<'a, B> {
 
     /// Create a new `BitmapSlice` starting from the specified `offset` into the current slice.
     fn slice_at(&self, offset: usize) -> Self {
-        RefSlice {
-            inner: self.inner,
+        BaseSlice {
+            inner: self.inner.clone(),
             base_offset: self.base_offset.wrapping_add(offset),
         }
     }
 }
 
-impl<'a, B> Clone for RefSlice<'a, B> {
-    fn clone(&self) -> Self {
-        RefSlice {
-            inner: self.inner,
-            base_offset: self.base_offset,
-        }
-    }
-}
-
-impl<'a, B> Copy for RefSlice<'a, B> {}
-
-impl<'a, B> Debug for RefSlice<'a, B> {
+impl<B> Debug for BaseSlice<B> {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         // Dummy impl for now.
         write!(f, "(bitmap slice)")
     }
 }
 
+/// A `BitmapSlice` implementation that wraps a reference to a `Bitmap` object.
+pub type RefSlice<'a, B> = BaseSlice<&'a B>;
+
+/// A `BitmapSlice` implementation that uses an `Arc` handle to a `Bitmap` object.
+pub type ArcSlice<B> = BaseSlice<Arc<B>>;
```

Comment:
Could you please also help to add some test cases for ArcSlice?

Reply:
The existing `test_ref_slice` unit test effectively covers `ArcSlice` as well (and all instantiations of `BaseSlice` basically), since the only difference is the smart pointer being used, which is transparent w.r.t. the test as long as `Deref` is properly implemented. Renamed `test_ref_slice` to `test_slice` to better express the generality.